gusto/diagnostics.py

```diff
@@ -39,6 +69,10 @@ def compute(self, state):
         """ Compute the diagnostic field from the current state"""
         pass
 
+    def _setup_solver(self, state):
+        """Setup solvers for calculating this diagnostic field"""
+        pass
```

Comment:
I don't understand the bit about passing the class constructor in to state - what would that look like? Also, your compute method takes ... args - do you mean to imply that different compute methods could take different arguments? How does that work?